subprojects/testing-base/src/testFixtures/groovy/org/gradle/testing/AbstractTestFrameworkIntegrationTest.groovy

```diff
@@ -169,7 +171,7 @@ abstract class AbstractTestFrameworkIntegrationTest extends AbstractIntegrationS
         createPassingFailingTest()
 
         when:
-        run testTaskName, '--tests', "${testSuite('SomeOtherTest')}.$passingTestCaseName"
+        run testTaskName, '--tests', "${testSuite('SomeOtherTest')}.${removeParentheses(passingTestCaseName)}"
```

Comment:
❌ Why do we remove parentheses here? Seems like we shouldn't be passing them in the first place.

Reply:
Let's say we have a method named `testMethod()`, under JUnit4 the "test case name" is "testMethod", under JUnit platform the "test case name" is "testMethod()", but we should always pass command line args as `--tests 'testMethod'`, that's why extra parentheses should be moved here.

Reply:
Why doesn't `--tests testMethod()` work?

Reply:
Because we're reusing [TestSelectionMatcher](https://github.com/gradle/gradle/blob/85275a02d6c1c06e2e81263c29b793ea2ec88946/subprojects/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/filter/TestSelectionMatcher.java#L26) class to make the filtering behaviour consistent. I'm a bit reluctant to change this matcher class (add `()` support) to adapt this behavior. This will make this PR more complex and we could improve it later.

Reply:
Okay, can you add an issue to investigate supporting `--tests` with parameterized JUnit 5 tests?

Reply:
Here it is: https://github.com/gradle/gradle/issues/4307